tests/prague/eip7480_data_section/test_data_opcodes.py

```diff
@@ -0,0 +1,254 @@
+                    code=(
+                        Op.CALLF[1]
+                        + Op.CALLF[2]
+                        + Op.CALLF[3]
+                        + Op.CALLF[4]
+                        + Op.PUSH1(1)
+                        + Op.PUSH0
+                        + Op.SSTORE
+                        + Op.STOP
+                    ),
+                    code_inputs=0,
+                    code_outputs=NON_RETURNING_SECTION,
+                    max_stack_height=2,
+                ),
+                Section.Code(
+                    code=(Op.PUSH2(offset) + Op.DATALOAD + Op.PUSH1(1) + Op.SSTORE + Op.RETF),
+                    code_inputs=0,
+                    code_outputs=0,
+                    max_stack_height=2,
+                ),
+                Section.Code(
+                    code=(dataloadn_op + Op.PUSH1(2) + Op.SSTORE + Op.RETF),
+                    code_inputs=0,
+                    code_outputs=0,
+                    max_stack_height=2,
+                ),
+                Section.Code(
+                    code=(Op.DATASIZE + Op.PUSH1(3) + Op.SSTORE + Op.RETF),
+                    code_inputs=0,
+                    code_outputs=0,
+                    max_stack_height=2,
+                ),
+                Section.Code(
+                    code=(
+                        Op.PUSH1(32)
+                        + Op.PUSH2(offset)
+                        + Op.PUSH0
+                        + Op.DATACOPY
+                        + Op.PUSH0
+                        + Op.MLOAD
+                        + Op.PUSH1(4)
+                        + Op.SSTORE
+                        + Op.RETF
+                    ),
+                    code_inputs=0,
+                    code_outputs=0,
+                    max_stack_height=3,
+                ),
+                Section.Data(data),
+            ],
+        ),
+        {0: 1, 1: answer, 2: answer, 3: datasize, 4: answer},
+    )
+
+
+CALL_SUCCEED_CONTRACTS: List[Tuple[str, Container, dict[int, int]]] = [
+    create_data_test("empty_zero", 0, 0),
+    create_data_test("short_zero", 0, 2),
+    create_data_test("exact_zero", 0, 32),
+    create_data_test("large_zero", 0, 64),
+    create_data_test("empty_32", 32, 0),
+    create_data_test("short_32", 32, 34),
+    create_data_test("exact_32", 32, 64),
+    create_data_test("large_32", 32, 96),
+    create_data_test("empty_23k", 0x5BFE, 0),
+    create_data_test("short_23k", 0x5BFE, 0x5C00),
+    create_data_test("exact_23k", 0x5BE0, 0x5D00),
+    create_data_test("large_23k", 0x2345, 0x5C00),
+]
+"""
+List of all EOF V1 Containers that simply need to succeed on execution.
+"""
+
+
+@pytest.mark.parametrize(
+    ["name", "container", "expected_storage"],
+    CALL_SUCCEED_CONTRACTS,
+    ids=[case[0] for case in CALL_SUCCEED_CONTRACTS],
+)
+def test_data_section_succeed(
+    state_test: StateTestFiller,
+    name: str,
+    container: Container,
+    expected_storage: dict[int, int],
+):
```

Comment:
```suggestion
@pytest.mark.parametrize(
    ["offset", "datasize"],,
    [
        pytest.param(0, 0, id="empty_zero"),
        pytest.param(0, 2, id="short_zero"),
        pytest.param(0, 32, id="exact_zero"),
        pytest.param(0, 64, id="large_zero"),
        pytest.param(32, 0, id="empty_32"),
        pytest.param(32, 34, id="short_32"),
        pytest.param(32, 64, id="exact_32"),
        pytest.param(32, 96, id="large_32"),
        pytest.param(0x5BFE, 0, id="empty_23k"),
        pytest.param(0x5BFE, 0x5C00, id="short_23k"),
        pytest.param(0x5BE0, 0x5D00, id="exact_23k"),
        pytest.param(0x2345, 0x5C00, id="large_23k"),
    ],
)
def test_data_section_succeed(
    state_test: StateTestFiller,
    offset: int,
    datasize: int,
):
```

I think the issue with this arrangement is that, even if you just want to fill a single vector, the container generation logic of all containers is executed.

My suggestion here is to bring `create_data_test` logic into `test_data_section_succeed`, and swap parameters `container` and `expected_storage` with `offset` and `datasize` in `test_data_section_succeed`.